### PR TITLE
Implement calcparams_pvsyst and tests

### LINKED ISSUE #470: Implement PVsyst single diode model
I've heard a number of requests for this capability, which we have in MATLAB but not here.  Requires calcparams_PVsyst and an example set of module_parameters.

### PR DESCRIPTION
**Pull request by cwhanse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python!

You may submit a pull request with your code at any stage of completion, however, before the code can be merged the following items must be addressed:

 - [x] Closes #470 
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Please don't hesitate to ask for help if you're unsure of how to accomplish any of the above. You may delete all of these instructions except for the list above.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

Implement `calcparams_pvsyst` for use with `singlediode`

### CODE REVIEW

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -1172,6 +1202,147 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+    temp_cell : numeric
+        The average cell temperature of cells within a module in C.
+
+    alpha_sc : float
+        The short-circuit current temperature coefficient of the
+        module in units of A/C.
+
+    gamma_ref : float
+        The diode ideality factor
+
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: missing mugamma documentation here. mugamma or mu_gamma?

File: pvlib/pvsystem.py [hunk 2]
```diff
@@ -307,6 +307,36 @@ def calcparams_desoto(self, effective_irradiance, temp_cell, **kwargs):
...
+        Parameters
+        ----------
+        effective_irradiance : numeric
+            The irradiance (W/m2) that is converted to photocurrent.
+
+        temp_cell : float or Series
+            The average cell temperature of cells within a module in C.
+
+        **kwargs
+            See pvsystem.calcparams_desoto for details
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: pvsyst, not desoto.

**Comment by cwhanse** [comment]
@mikofski would appreciate your review of `calcparams_pvsyst`

File: pvlib/pvsystem.py [hunk 3]
```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+        
+    R_sh_0 : float
+        The shunt resistance at zero irradiance conditions, in ohms.
+        
+    R_s : float
+        The series resistance at reference conditions, in ohms.
+
+    cells_in_series : integer
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: `Ns` would seem more consistent with the desoto model and the `nNsVth` variable.

File: pvlib/pvsystem.py [hunk 4]
```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+        in amperes.
+        
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions,
+        in amperes.
+        
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+        
+    R_sh_0 : float
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: `R_sh_ref` seems more consistent with other variable names.

File: pvlib/pvsystem.py [hunk 5]
```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+        The shunt resistance at zero irradiance conditions, in ohms.
+        
+    R_s : float
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: R_s_ref seems more consistent with other variables at reference conditions.

File: pvlib/pvsystem.py [hunk 6]
```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+        The average cell temperature of cells within a module in C.
+
+    alpha_sc : float
+        The short-circuit current temperature coefficient of the
+        module in units of A/C.
+
+    gamma_ref : float
+        The diode ideality factor
+
+    mugamma : float
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: Perhaps `mu_gamma`?

File: pvlib/pvsystem.py [hunk 7]
```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+
+    temp_ref : float (optional, default=25)
+        Reference cell temperature in C.
+
+    Returns
+    -------
+    Tuple of the following results:
+
+    photocurrent : numeric
+        Light-generated current in amperes at irradiance=S and
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: Not sure about `irradiance=S` here and below.

**Comment by thunderfish24** [comment]
@cwhanse Are you not planning to integrate the pvsyst model into `ModelChain` in this PR?

**Comment by mikofski** [comment]
Baby steps @thunderfish24. Will this PR be incomplete without it?


* Haste makes waste.
* Bird in the hand is worth two in the bush
* Don't put all your eggs in one basket

**Comment by thunderfish24** [comment]
@mikofski This was by no means a criticism. Rather, I was just wondering if complete integration was expected right away. LIkewise, #478 is much more easily implemented if I can worry about the `ModelChain` integration as a followup step. 

@cwhanse I do propose this PR adds a usage example in a subsection of `docs/tutorials/pvsystem.ipynb`.

**Comment by mikofski** [comment]
I'm so sorry @thunderfish24 my comment was snarky. It was a fair question.

**Comment by thunderfish24** [comment]
@mikofski No worries mate! My question could have been phrased better, and it's often easy to misinterpret typings when we are not face-to-face.

**Comment by cwhanse** [comment]
@mikofski @thunderfish24 yes I'll open a new Issue and PR for extending `ModelChain` for new single diode models.

File: pvlib/pvsystem.py (hunk 1 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: added.  changed to `mu_gamma`

File: pvlib/pvsystem.py [hunk 8]
```diff
@@ -1172,6 +1202,147 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+        cell temperature=Tcell.
+
```
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: I edited the output description (and in `pvsystem.calcparams_desoto`) to remove the phrase 'at irradiance....=Tcell' I think the context is clear.

File: pvlib/pvsystem.py [hunk 9]
```diff
@@ -1172,6 +1202,147 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+    # Boltzmann constant in J/K
+    k = 1.3806488e-23
+
+    # elementary charge in coulomb
+    q = 1.6021766e-19
+    
+    # reference temperature
+    Tref_K = temp_ref + 273.15
+    Tcell_K = temp_cell + 273.15
+
```
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: these are local variables, so I think its OK that they are different in style.

[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: So I changed them as you suggested :)

File: pvlib/pvsystem.py (hunk 3 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: I agree but we use `cells_in_series` as the key for the SAPM parameters, and some have a preference for descriptive variable names.

File: pvlib/pvsystem.py (hunk 5 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: I agree. But here we run into an unfortunate linkage to the SAM database for the CEC model parameters.  `R_s` is the heading in that file, which is used to define the keys for the module parameter dict. Although in my opinion `R_s` for PVsyst is a different parameter than `R_s` for De Soto or CEC (because parameter values are determined jointly, they are intrinsically linked and can't be regarded as correct for a different model) I do think it's useful to maintain a common nomenclature.

I think changing `R_s` to `R_s_ref` will involve some work in `pvsystem.retrieve_sam` so I'm not going to take that on in this PR.

File: pvlib/pvsystem.py (hunk 6 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: yes

File: pvlib/pvsystem.py (hunk 7 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: nuked

File: docs/sphinx/source/whatsnew/v0.6.0.rst [hunk 10]
```diff
@@ -13,6 +13,7 @@ Enhancements
 * Add sea surface albedo in irradiance.py (:issue:`458`)
 * Implement first_solar_spectral_loss in modelchain.py (:issue:`359`)
 * Clarify arguments Egref and dEgdT for calcparams_desoto (:issue:`462`)
+* Add pvsystem.calcparams_pvsyst to compute values for the single diode equation using the PVsyst v6 model.
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.6.0.rst]: (:issue:`470`)

File: pvlib/pvsystem.py (hunk 1 above)
Comment (wholmgren) [reply pvlib/pvsystem.py]: it appears that the change was not applied here.

File: pvlib/test/test_pvsystem.py [hunk 11]
```diff
@@ -174,6 +174,23 @@ def cec_module_params(sam_data):
     return module_parameters
 
 
+@pytest.fixture(scope="session")
+def pvsyst_module_params():
+    module_parameters = {}
+    module_parameters['gamma_ref'] = 1.05
+    module_parameters['mugamma'] = 0.001
```
Comment (wholmgren) [inline pvlib/test/test_pvsystem.py]: test fixture and the two tests below need mugamma --> mu_gamma

**Comment by cwhanse** [comment]
I changed a setting in Spyder, hopefully that improves the white space issue.  Test failures are for `read_tmy3`

File: pvlib/pvsystem.py [hunk 12]
```diff
@@ -307,6 +307,36 @@ def calcparams_desoto(self, effective_irradiance, temp_cell, **kwargs):
...
+        Parameters
+        ----------
+        effective_irradiance : numeric
+            The irradiance (W/m2) that is converted to photocurrent.
+
+        temp_cell : float or Series
+            The average cell temperature of cells within a module in C.
+
+        **kwargs
+            See pvsystem.calcparams_pvsyst for details
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: and delete ``**kwargs`` from the docstring.

**Comment by wholmgren** [comment]
I think this should be merged after resolving the minor issues around ``**kwargs``

File: pvlib/pvsystem.py [hunk 13]
```diff
@@ -307,6 +307,36 @@ def calcparams_desoto(self, effective_irradiance, temp_cell, **kwargs):
         
         return calcparams_desoto(effective_irradiance, temp_cell, **kwargs)
 
+    def calcparams_pvsyst(self, effective_irradiance, temp_cell, **kwargs):
```
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: done

File: pvlib/pvsystem.py (hunk 12 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: done

File: pvlib/pvsystem.py [hunk 14]
```diff
@@ -1011,28 +1040,21 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
 
     saturation_current : numeric
-        Diode saturation curent in amperes at irradiance
-        S and cell temperature Tcell.
+        Diode saturation curent in amperes
 
     resistance_series : float
-        Series resistance in ohms at irradiance S and cell temperature
-        Tcell.
+        Series resistance in ohms
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: While passing through this territory, seems like this should be changed to "numeric" from "float".

File: pvlib/pvsystem.py [hunk 15]
```diff
@@ -302,11 +302,40 @@ def calcparams_desoto(self, effective_irradiance, temp_cell, **kwargs):
...
+        """
+
+        kwargs = _build_kwargs(['gamma_ref', 'mu_gamma', 'I_L_ref', 'I_o_ref',
+                                'R_sh_ref', 'R_sh_0', 'R_sh_exp',
+                                'R_s', 'alpha_sc', 'EgRef',
+                                'irrad_ref', 'temp_ref',
+                                'cells_in_series'],
+                                self.module_parameters)
+
+        return calcparams_pvsyst(effective_irradiance, temp_cell, **kwargs)
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: This could be done in a separate issue (see #476), but I think the `PVSystem` methods should scale the outputs to the `modules_per_string` and `strings_per_inverter`.

File: pvlib/pvsystem.py [hunk 16]
```diff
@@ -1164,14 +1184,151 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+
+    See Also
+    --------
+    calcparams_desoto
+    singlediode
+
+    '''
+
+    # Boltzmann constant in J/K
+    k = 1.3806488e-23
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: Appears to differ a bit from `scipy.constants.Boltzmann`, which is `1.38064852e-23`

File: pvlib/pvsystem.py [hunk 17]
```diff
@@ -1164,14 +1184,151 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+    nNsVth = gamma * k / q * cells_in_series * Tcell_K
+
+    IL = effective_irradiance / irrad_ref * \
+              (I_L_ref + alpha_sc * (Tcell_K - Tref_K))
+
+    I0 = I_o_ref * ((Tcell_K / Tref_K) ** 3) * \
+          (np.exp((q * EgRef) / (k * gamma) * (1 / Tref_K - 1 / Tcell_K)))
+
+    Rsh_tmp = (R_sh_ref - R_sh_0 * np.exp(-R_sh_exp)) / (1.0 - np.exp(-R_sh_exp))
+    Rsh_base = np.maximum(0.0, Rsh_tmp)
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: In looking over http://files.pvsyst.com/help/pvmodule_rshexp.htm, it is not clear to me why `Rsh_base` is not simply `R_sh_ref`. What am I missing?

File: pvlib/pvsystem.py [hunk 18]
```diff
@@ -1164,14 +1184,151 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+
+    # reference temperature
+    Tref_K = temp_ref + 273.15
+    Tcell_K = temp_cell + 273.15
+
+    gamma = gamma_ref + mu_gamma * (Tcell_K - Tref_K)
+    nNsVth = gamma * k / q * cells_in_series * Tcell_K
+
+    IL = effective_irradiance / irrad_ref * \
+              (I_L_ref + alpha_sc * (Tcell_K - Tref_K))
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: Optimization possibility: `Tcell_K - Tref_K` is computed twice.

File: pvlib/pvsystem.py [hunk 19]
```diff
@@ -1164,14 +1184,151 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+
+    I0 = I_o_ref * ((Tcell_K / Tref_K) ** 3) * \
+          (np.exp((q * EgRef) / (k * gamma) * (1 / Tref_K - 1 / Tcell_K)))
+
+    Rsh_tmp = (R_sh_ref - R_sh_0 * np.exp(-R_sh_exp)) / (1.0 - np.exp(-R_sh_exp))
+    Rsh_base = np.maximum(0.0, Rsh_tmp)
+
+    Rsh = Rsh_base + (R_sh_0 - Rsh_base) * \
+              np.exp(-R_sh_exp * effective_irradiance / irrad_ref)
```
Comment (thunderfish24) [inline pvlib/pvsystem.py]: Optimization possibility: `effective_irradiance / irrad_ref` is computed twice.

File: pvlib/test/test_pvsystem.py [hunk 20]
```diff
@@ -386,6 +403,35 @@ def test_calcparams_desoto(cec_module_params):
...
+                                  EgRef=pvsyst_module_params['EgRef'])
+
+    assert_series_equal(np.round(IL, 3), pd.Series([0.0, 4.8200], index=times))
+    assert_series_equal(np.round(I0, 3),
+                        pd.Series([0.0, 1.47e-7], index=times))
+    assert_allclose(Rs, 0.500)
+    assert_series_equal(np.round(Rsh, 3),
+                        pd.Series([1000.0, 305.757], index=times))
+    assert_series_equal(np.round(nNsVth, 4),
+                        pd.Series([1.6186, 1.7961], index=times))
```
Comment (thunderfish24) [inline pvlib/test/test_pvsystem.py]: My preference is to compute the expected values directly in the test. I'm curious if you have particular reasons not to do that?

File: pvlib/test/test_pvsystem.py [hunk 21]
```diff
@@ -386,6 +403,35 @@ def test_calcparams_desoto(cec_module_params):
     assert_allclose(nNsVth, 0.473)
 
 
+def test_calcparams_pvsyst(pvsyst_module_params):
```
Comment (thunderfish24) [inline pvlib/test/test_pvsystem.py]: Not critical, but you might consider a simple "API" test that passes the five computed coefficients into `singlediode` just to make sure the computation runs. This makes sure that pvsyst gets updated if the API changes.

File: pvlib/pvsystem.py (hunk 15 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: We've chosen not to scale inside `dc_model` functions to offer flexibility, e.g., model a system with strings of different lengths, to implement module-to-module mismatch models, etc. Granted some of these capabilities are not presently in pvlib.

File: pvlib/pvsystem.py (hunk 14 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: Cut and paste inheritance?  I don't think there is a difference here. `numeric` would admit `long`, `complex` and `int` types as well as `float`.  Probably should change to `float` unless there's a reason otherwise.

File: pvlib/pvsystem.py (hunk 16 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: Good catch.  That's a typo.

File: pvlib/pvsystem.py (hunk 17 above)
Comment (cwhanse) [reply pvlib/pvsystem.py]: You're not, PVsyst documentation is missing that definition of `Rsh_base`. Ken Sauer and Thomas Roessler are my sources, they got it verbally from PVsyst.

File: pvlib/test/test_pvsystem.py (hunk 20 above)
Comment (cwhanse) [reply pvlib/test/test_pvsystem.py]: I don't other than it was easy (for me) to compute the values in MATLAB and hardcode them here.  My way is certainly less transparent.

File: pvlib/pvsystem.py (hunk 14 above)
Comment (thunderfish24) [reply pvlib/pvsystem.py]: I'm good with float instead of numeric, because that seems to be most accurate.

File: pvlib/pvsystem.py [hunk 22]
```diff
@@ -1164,14 +1184,151 @@ def calcparams_desoto(effective_irradiance, temp_cell,
...
+                      R_sh_exp=5.5,
+                      EgRef=1.121,
+                      irrad_ref=1000, temp_ref=25):
+    '''
+    Calculates five parameter values for the single diode equation at
+    effective irradiance and cell temperature using the PVsyst v6
+    model described in [1,2,3]. The five values returned by calcparams_pvsyst
+    can be used by singlediode to calculate an IV curve.
+
+    Parameters
```
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: I'm pretty sure the function will work on array or Series inputs. Is the issue the type description `float` implies a singleton?

We aren't using `numeric` or `float` consistently in the comments.  For me to fix that, requires that I learn the distinction.  New issue, I think.

File: pvlib/test/test_pvsystem.py (hunk 21 above)
Comment (cwhanse) [reply pvlib/test/test_pvsystem.py]: Fair point. I'm not going to put a new test in here because I don't clearly see how to separate issues being tested, and to avoid redundant tests. The current `test_calcparams_pvsyst` verifies correct output of `pvsystem.calcparams_pvsyst`. Adding a call to `singlediode` within `test_calcparams_pvsyst` adds a new purpose (API check) and I prefer to keep tests to a single purpose. Adding a new test that calls `calcparams_pvsyst` and then `singlediode` would duplicate the test of `PVSystem.calcparams_pvsyst` method.

Do you have an example in the `campanelli` pull request?

File: pvlib/pvsystem.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/pvsystem.py]: I've used ``numeric`` for input that can be scalar, ``np.array``, or ``pd.Series``. ``float`` is specific to scalar. Whether or not this is a good idea could be discussed in a separate issue, if necessary.

[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/pvsystem.py]: for more context, note that some functions require "array-like" (``np.array`` or ``pd.Series``) and others require ``pd.Series``.

File: pvlib/pvsystem.py (hunk 16 above)
Comment (wholmgren) [reply pvlib/pvsystem.py]: ref #483.

File: pvlib/pvsystem.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: So, `numeric` unless there's a reason to specify a scalar `float`?

File: pvlib/test/test_pvsystem.py [hunk 23]
```diff
@@ -174,6 +174,23 @@ def cec_module_params(sam_data):
     return module_parameters
 
 
+@pytest.fixture(scope="session")
```
Comment (wholmgren) [inline pvlib/test/test_pvsystem.py]: not a big deal, but if we're still making changes this should not have ``scope="session"``

File: pvlib/test/test_pvsystem.py (hunk 20 above)
Comment (wholmgren) [reply pvlib/test/test_pvsystem.py]: I'm confused. I don't understand how general test values can be computed inside a test without duplicating the function itself.

File: pvlib/pvsystem.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/pvsystem.py]: or unless there's a reason to specify ``array-like`` or ``Series``.

[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/pvsystem.py]: I looked at both `calcparams_desoto` and `calcparams_pvsyst` and I'm going to leave the docstrings as is _for now_. The typical use of either function is to pass model parameters for one module and a list of conditions (irradiance and temperature). That use is reflected in the docstring's use of `numeric` and `float`, where each model parameter is a single value for a module.  The functions are more flexible that this use case.

If we want to overhaul the docstrings, let's open a new issue for it.

File: pvlib/test/test_pvsystem.py (hunk 23 above)
Comment (cwhanse) [reply pvlib/test/test_pvsystem.py]: I copied that.  `scope="module"`?

Comment (wholmgren) [reply pvlib/test/test_pvsystem.py]: no scope argument. ``pytest.fixture`` and nothing else

Comment (wholmgren) [reply pvlib/test/test_pvsystem.py]: The reasoning for this is that this function returns a mutable dictionary. In principle, one test could inadvertently modify the dictionary, and a second test could pass or fail because of that modification. ``@pytest.fixture`` will call the function and generate a new dictionary each time the data is needed. Adding ``scope="module"`` or ``scope="session"`` instructs ``pytest`` to recreate the data only once per test module (the fixture could be imported into a new module and data would be regenerated) or once per test session (this defeats a large part of the purpose of fixtures).

Comment (cwhanse) [reply pvlib/test/test_pvsystem.py]: OK. I'll remove the other instances of `scope=session` from fixtures for CEC and SAPM module parameters, but leave the instance that uses `pvsystem.retrieve_sam`

**Comment by cwhanse** [comment]
Looks like errors related to pulling tmy3 data from NREL's site.

File: pvlib/pvsystem.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (thunderfish24) [reply pvlib/pvsystem.py]: I think an issue is wise in order to define what we mean to ourselves/users and make everything consistent.

File: pvlib/test/test_pvsystem.py (hunk 21 above)
Comment (thunderfish24) [reply pvlib/test/test_pvsystem.py]: No, I haven't written any tests yet and I admit that I am not familiar with this particular mocker setup. If a mock call can verify the API, then that would be great. I don't mean to hold up this PR and I'll look into it further with the Campanelli model.

File: pvlib/test/test_pvsystem.py (hunk 20 above)
Comment (thunderfish24) [reply pvlib/test/test_pvsystem.py]: Some view including a separate (re)computation as redundant, others view it as transparent. In this case I don't think any of the computations require an implicit solver, so where the values come from should not leave too much to question.

**Comment by wholmgren** [comment]
@thunderfish24 thanks for your careful review -- it's a big help. 

@cwhanse ready to merge?

**Comment by cwhanse** [comment]
Yes, ready to merge.

File: docs/sphinx/source/whatsnew/v0.6.0.rst [hunk 24]
```diff
@@ -13,6 +13,7 @@ Enhancements
 * Add sea surface albedo in irradiance.py (:issue:`458`)
 * Implement first_solar_spectral_loss in modelchain.py (:issue:`359`)
 * Clarify arguments Egref and dEgdT for calcparams_desoto (:issue:`462`)
+* Add pvsystem.calcparams_pvsyst to compute values for the single diode equation using the PVsyst v6 model (:issue:'470')
```
Comment (mikofski) [inline docs/sphinx/source/whatsnew/v0.6.0.rst]: please change single spaces to backticks - @wholmgren or @cwhanse , can you do this directly in master, thx
